front/app/modules/commercial/project_folders/permissions/rules.ts

```diff
@@ -8,24 +8,42 @@ import { isAdmin } from 'services/permissions/roles';
 import {
   canAccessRoute,
   isAdminRoute,
-  MODERATOR_ROUTES,
+  MODERATOR_INDEX_ROUTES,
 } from 'services/permissions/rules/routePermissions';
 import { IUser } from 'services/users';
 import { IAppConfigurationData } from 'services/appConfiguration';
+import { removeLocale } from 'utils/cl-router/updateLocationDescriptor';
 
 const canUserAccessAdminFolderRoute = (
```

Comment:
This is firing on every single route change in the app related to the admin section. If you console log this function, this is how many times it fires when I click "manage platform" from the main page, as a moderator:

![Screen Shot 2022-06-14 at 07 48 09](https://user-images.githubusercontent.com/3614128/173502593-443ab51a-f4e4-463d-8414-cf669fc5b7a9.png)

As such I think it should be removed. For most use cases it will return "accurately" because it does the same `canAccessRoute(item, user, tenant)` check at the beginning, but that means we're always doing two permissions checks on every route change.

Reply:
Yes, it's not extended correctly.

Reply:
We should probably make it the same as for project moderators.

Reply:
Where is the code that deals with project moderators?